convert2rhel/systeminfo.py

```diff
@@ -320,38 +325,62 @@ def get_enabled_rhel_repos(self):
+        This method will try to retrieve a web page on the Red Hat network that
+        we know to exist (http://static.redhat.com/test/rhel-networkmanager.txt).
+        If we can successfully retrieve that page, validate that the status_code
+        is 200, and assert that the content of the response is the string 'OK',
+        then we decide we are connected to the internet.
+
+        We check a web page because we will need working https to retrieve
+        packages from Red Hat infrastructure during the conversion.
 
         .. warnings::
             We might have some problems with this if the host machine is using
-            a NAT gateway to route the outbound requests
-
-        .. seealso::
-            * Comparison of different methods to check internet connections: https://stackoverflow.com/a/33117579
-            * Redirecting IP addresses: https://superuser.com/questions/954665/how-to-redirect-route-ip-address-to-another-ip-address
-
-        :param host: The host to establish a connection to. Defaults to "8.8.8.8".
-        :type host: str
-        :param port: The port for the connection. Defaults to 53.
-        :type port: int
-        :param timeout: The timeout in seconds for the connection. Defaults to 3.
-        :type port: int
-        :return: Return boolean indicating whether or not we have internet access.
+            a NAT gateway to route the outbound requests to any other service.
+
+            DNS could also be used to redirect the URL we test to another address.
+
+        :return: Return boolean indicating whether or not we have internet
+            access.
         :rtype: bool
         """
-        try:
-            self.logger.info("Checking internet connectivity using host '%s' and port '%s'." % (host, port))
-            socket.setdefaulttimeout(timeout)
-            socket.socket(socket.AF_INET, socket.SOCK_STREAM).connect((host, port))
-            return True
-        except (socket.timeout, socket.error):
-            self.logger.warning("Couldn't connect to the host '%s'." % host)
-            return False
+        retries = 0
+        is_reachable = False
+        self.logger.info(
+            "Checking internet connectivity using address '%s'.",
+            CHECK_INTERNET_CONNECTION_ADDRESS,
+        )
+        while retries <= CHECK_INTERNET_CONNECTION_RETRIES:
+            try:
+                response = urllib.request.urlopen(CHECK_INTERNET_CONNECTION_ADDRESS)
+            except urllib.error.URLError as err:
+                self.logger.debug("Faild to retrieve data from host, reason: %s", err.reason)
+                break
+
+            status_code = response.getcode()
+            text_response = response.read().decode("utf-8")
+            response.close()
+
+            if status_code == 200 and text_response.upper() == "OK":
+                self.logger.info("Internet connection available.")
+                is_reachable = True
+                break
```

Comment:
I don't see why we need to check the text response or status code. We're not checking the status of the service, we're checking that the internet connectivity is working. 

If we only test status code, we would have to test all status codes, in which the if-statement becomes pointless. If we don't then we're naive. The server could have moved the endpoint, changed the format, etc.

All we need to do is ensure we can get a response from the server. Content or status does not matter. In the event that the server is completely down we need to be able to handle that scenario as well

Reply:
I implemented that part because it was part of the acceptance criteria defined in the ticket. 

> Download and verify the content of the file returned from the pre-defined URL. In the above case, it's an XML file containing some repo information.

I vaguely remember that we discussed that, in one cabal session, that it would be good to verify the content of the response. Regardless of that, I can change this and remove this condition if you think that's appropriate.

Reply:
If we want to verify the contents of the file the workflow should be as follows:

1. Verify internet connectivity
2. Read contents of XML file
    1. If contents can be read, get repo information
    2. If contents can not be read, continue as normal

Reply:
> If we want to verify the contents of the file the workflow should be as follows (oversimplified):
> 
>     1. Verify internet connectivity
>        
>        1. If internet connectivity exists, continue function
>        2. If internet connectivity does not exist, return false

Yeah… This is exactly what the code is doing right now. If the connection can be established through the `urllib.request.urlopen()`, then we progress with `1.`, otherwise, if the exception is raised `urllib.error.URLError`, meaning that the connection was refused (no connection, dnsmasq redirects, server offline, nat gateway redirects, firewalls, url not being accessible etc… Anything that can refuse a request) then we go with `2.`. 

>     2. Read contents of XML file
>        
>        1. If contents can be read, get repo information
>        2. If contents can not be read, continue as normal

That's what we are doing. The exception is that we are not using XML files to check repo availability, but a static file that just returns 'OK' in case of being reachable. I don't see a difference between using the example URL that just returns an XML and this one, that just returns a piece of information. To be honest, any internet check can be bypassed by an experienced network admin, but I tend to believe they won't do that on purpose.

Sorry, but that's still not clear to me what is expected with that comment, if you could detail a bit more on why we need to remove, I will be glad… Maybe I'm just missing some point here.

Note: I'm not against removing this condition, I agree that status code here might not be the best information for a check, might as well make sense to remove it. Just asking for a bit more clarification on what you expect to see in this change.

Reply:
> This is exactly what the code is doing right now

No the code is only asserting that the connection is good if the content is OK and status code is 200. Otherwise it fails. That's what I'm trying to establish is an issue here as we tie the connection to the service rather than if we can connect

The thing I'm trying to say is that we should not establish if we have connectivity based on status code or content

Reply:
Now it's more clear, I see your point now, thanks :) 

So, simple removing this condition and letting the `try-except` the way it is right now should be sufficient for that.